code/modules/jobs/job_types/security_officer.dm

```diff
@@ -133,7 +133,7 @@ GLOBAL_LIST_EMPTY(security_officer_distribution)
 	var/list/partners = list()
 	for (var/officer_ref in distribution)
 		var/mob/partner = locate(officer_ref)
-		if (!istype(partner))
+		if (!istype(partner) || distribution[officer_ref] != department)
```

Comment:
```suggestion
		if (!istype(partner) || distribution[officer_ref] != department))
```
Missed a bracket?

Reply:
Nope. The original code is `if(() || [])`
Your variant is `if((() || []))` which is one too many parenthesis. The original code was correct.